sklearn/utils/_array_api.py

```diff
@@ -119,14 +119,22 @@ def _isdtype_single(dtype, kind, *, xp):
         return dtype == kind
 
 
+def supported_float_dtypes(xp):
+    """Supported floating point types for the namespace"""
+    if hasattr(xp, "float16"):
+        return (xp.float64, xp.float32, xp.float16)
+    else:
+        return (xp.float64, xp.float32)
```

Comment:
Nit: 

```suggestion
        return {xp.float64, xp.float32, xp.float16}
    else:
        return {xp.float64, xp.float32}
```

Reply:
I am pretty sure that tuples are always faster / less overhead for most operations when they have low cardinality. But fair enough, I am fine with either :)

Reply:
Okay, lets go with tuples.